src/Woman.java

```diff
@@ -0,0 +1,36 @@
+public class Woman extends Person{
+    private String maidenlastName;
+
+    public Woman(String firstName, String lastName, int age) {
+        super(firstName, lastName, age);
+        this.maidenlastName = lastName;
+    }
+
+    @Override
+    public boolean isRetired() {
+        return (this.getAge()>=60);
+    }
+
+    public void registerPartnership(Person partner){
+        if (getPartner() == null) {
+            super.setLastName(partner.getLastName());
+
+        }
```

Comment:
It's not clear what are you doing here. You need to set partner field and set partner for partner

Reply:
here I set new last name for woman after her got married

Reply:
I was meaning that you need to do `this.partner.partner = this;` because you set it only from one object to another but not for both. Partnership it's bidirectional connection